pkgs/development/tools/misc/gf/default.nix

```diff
@@ -0,0 +1,63 @@
+{ lib
+, stdenv
+, makeWrapper
+, fetchFromGitHub
+, libX11
+, pkg-config
+, gdb
+, freetype
+, freetypeSupport ? false
+, extensionFolder ? null
+}:
+
+stdenv.mkDerivation rec {
+  pname = "gf";
+  version = "unstable-2022-06-22";
+
+  src = fetchFromGitHub {
+    repo = "gf";
+    owner = "nakst";
+    rev = "e0d6d2f59344f853a4a204d5313db6b6a5e5de7d";
+    sha256 = "01fln4wnn1caqr4wa1nhcp0rqdx5m5nqyn2amvclp5hhi3h3qaiq";
+  };
+
+  nativeBuildInputs = [ makeWrapper pkg-config ];
+  buildInputs = [ libX11 gdb ]
+    ++ lib.optional freetypeSupport freetype;
+
+  inherit extensionFolder;
+
+  postPatch = ''
+    patchShebangs build.sh
+
+    if [ ! -z "$extensionFolder" ]; then
+      echo adding extensions from $extensionFolder;
+      cp -r "$extensionFolder"/* .
+    fi
+  '';
```

Comment:
```suggestion
  postPatch = lib.forEach extensions (ext: ''
      cp ${ext} .
  )'';
```

Reply:
Thanks for your suggestions. This helps a lot, however with this approach the filenames are prefixed with the nix-hash (e.g. 3p75yihvn1k56ax019i0grqvwg4fja2s-extensions.cpp). Is there a simple way to parse the original filename? Simply checking for the "-" will not work if the filename also has a dash in it. Otherwise the preprocessor magic does not recognize the files.

Reply:
Maybe `cp ${ext} ./${ext.name}` should help...

Reply:
Awesome, thanks for your input. It looks much cleaner now.